### ISSUE TITLE
Implement Maker / Taker Market Fees Flag

### ISSUE THREAD

**Issue by bytemaster** [cryptonomex/graphene#468]
Under this proposal, an asset issuer could enable a flag on their asset that would not charge any market fees for open orders.  The market fee would only be charged to the taker (the individual whose new order is actively being matched).

This would incentivise people to leave orders on the books (it is cheaper).  The only person who would pay a fee would be the one who wants immediate liquidity.

**Comment by theoreticalbts** [comment]
Assigning to OP to clarify two questions:
- Is this change something we want to do for the upcoming hardfork (i.e. #445 ), or postpone for a later hardfork?
- Is this change something which requires a separate proposal to be approved, or is this considered part of the worker proposal which funded #445 ?

When these questions are answered, feel free to remove "question" label and assign (or unassign) the issue.

**Comment by NaturalCoder** [comment]
Excellent guys, this will be a great incentive for liquidity. I think this should be considered an little change on the previous proposal since is almost the same, IMO we should not lose time approving all over again just because of that.

**Comment by bytemaster** [comment]
Since this feature does nothing unless the asset issuer updates the asset flags to enable this feature, we can leave it to the committee to decide (as this is a fee related question).  In this way the community (via committee) has control.

Whether or not this gets included in the current hard fork depends upon schedule. I do not think we should delay releasing the other fixes for this.

**Comment by bytemaster** [comment]
The following items still need to be done before this can be merged:
1. add unit tests for the new flags / fee calculations
2. ensure that initial assets start out with charge_maker_fee permission set to true and flag set to true (to match old behavior)
3. ensure that the wallet create asset initializes new assets with permission to charge maker fee, but the flag set to false.

**Comment by NaturalCoder** [comment]
About that, Isn't better instead of a flag, two fees adjustment, something
like taker fee and maker fee... I'm just thinking this could be needed on
UIA.

**Comment by theoreticalbts** [comment]
OK, on further discussion, we want to use a different approach.  Using the extension mechanism we can add a new optional field for taker fee; this will substantially simplify the hardfork logic and save the permission bit.

As a bonus it will let us do maker / taker fees independently as @NaturalCoder suggests.

**Comment by theoreticalbts** [comment]
Based on Mumble discussion, we are going to incentivize makers by issuing a new asset which will be bought and destroyed by the system with future market fees.

This will require a substantial redesign and a separate worker proposal, which should probably be a separate ticket.  When that worker proposal is completed, we will close this ticket.

**Comment by bytemaster** [comment]
Closing this ticket in favor of creating a new ticket with more details for a worker proposal.

**Comment by theoreticalbts** [comment]
Based on the above comment, I think the intention was to close this ticket, so I'm closing it.
